src/Admin.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+/**
+ * @file
+ * Contains \Netzstrategen\Varnish\Admin.
+ */
+
+namespace Netzstrategen\Varnish;
+
+/**
+ * Administrative back-end functionality.
+ */
+class Admin {
+
+  /**
+   * @implements admin_init
+   */
+  public static function init() {
+    // Confirm manual purge.
+    if (isset($_GET['varnish_flush_all']) && check_admin_referer(Plugin::PREFIX)) {
+      add_action('admin_notices', __CLASS__ . '::admin_notices_purged');
+    }
+
+    // Require custom permalinks to be set up.
+    if (!get_option('permalink_structure') && current_user_can('manage_options')) {
+      add_action('admin_notices', __CLASS__ . '::admin_notices_permalinks');
+    }
+
+    // Output manual purge button in admin bar.
+    if (static::hasPurgeAccess()) {
+      add_action('admin_bar_menu', __CLASS__ . '::admin_bar_menu'), 100);
+    }
+    // Output manual purge button on admin dashboard.
+    add_action('activity_box_end', __CLASS__ . '::activity_box_end', 100);
```

Comment:
Is ensured this is visible to admins with the correct privileges only? Probably it makes sense to move it into the `hasPurgeAccess` check as well?

Reply:
The meta box outputs a message about missing access privileges currently. We probably want to revisit the UI/UX in the future, but I don't think we should do that as part of this PR.